### DISCUSSION TITLE
Invoke C# from JavaScript - WASM

### DISCUSSION THREAD

**Discussion by TonyK-FmGroup** [unoplatform/uno#8615] [Q&A] [answered]
I've been following the tutorial here: https://platform.uno/docs/articles/interop/wasm-javascript-1.html?

My WASM projects is called `UnoJSTryout.Wasm` and I have a class `JsNotify`  in namespace `UnoJSTryout`.

In the javascript I then have this:

```
notifyPropertyChangedMethod = Module.mono_bind_static_method(
  "[UnoJSTryout.Wasm] UnoJsTryout.JsNotify:NotifyPropertyChanged");
```

The resulting exception in the browser says, "dotnet.js:1 
       Uncaught Error: Could not find class: UnoJsTryout:JsNotify in assembly UnoJSTryout.Wasm
    at Object.resolve_method_fqn (:49313/package_15bdf1460470e426cc7aa0f69fa0494a65473cf9/dotnet.js:1:192570)
    at Object.bind_static_method (:49313/package_15bdf1460470e426cc7aa0f69fa0494a65473cf9/dotnet.js:1:193126)
    at :49313/package_15bdf1460470e426cc7aa0f69fa0494a65473cf9/myapp.js:1:38"

Here's a snip of my class definition:
<img width="729" alt="image" src="https://user-images.githubusercontent.com/103487880/165123671-00acb65d-424a-43ea-a085-c9ce7379c175.png">

I think I've missed something obvious but I've not (yet) been able to find it.

**Comment by jeromelaban** [comment, marked as answer]
It could be a linker related issue: https://platform.uno/docs/articles/features/using-il-linker-webassembly.html. Is this class in the main project or in a separate project?

**Reply by TonyK-FmGroup** [reply]
The class is in the shared project. UnoJsTryout.Shared

**Reply by TonyK-FmGroup** [reply]
Thank you for the link! Adding this line to the LinkerConfig.xml has fixed my problem! 

```<assembly fullname="UnoJsTryout.Wasm"/>```
